src/wui/minimap.cc

```diff
@@ -70,11 +70,18 @@ bool MiniMap::View::handle_mousepress(const uint8_t btn, int32_t x, int32_t y) {
 	return true;
 }
 
-void MiniMap::View::set_zoom(int32_t z) {
+void MiniMap::View::set_zoom(const bool zoom) {
 	const Widelands::Map& map = ibase_.egbase().map();
-	set_size((map.get_width() * z), (map.get_height()) * z);
+	set_size(map.get_width() * scale_map(map, zoom), map.get_height() * scale_map(map, zoom));
 }
 
+bool MiniMap::View::can_zoom() {
+	const Widelands::Map& map = ibase_.egbase().map();
+	if (scale_map(map, true) == 1 || map.get_height() * scale_map(map, true) > ibase_.get_h() - 40) {
```

Comment:
40 is the size of the window border + minimap buttons?

Reply:
40px are the window borders. The buttons would be an additional 20. But I reckoned we could sacrifice 10px of the top and 10px of the bottom border too a zoomed map.

Reply:
The border thickness can be calculated by using `get_w() - get_inner_w()` from the UI::Panel class. I'd prefer to keep this dynamic to be stable whenever we change something in the basic Window layout.